compiler/AST/primitive.cpp

```diff
@@ -281,9 +281,11 @@ static Type*
 returnInfoEndCount(CallExpr* call) {
   static Type* endCountType = NULL;
   if (endCountType == NULL) {
-    forv_Vec(TypeSymbol, ts, gTypeSymbols) {
-      if (!strcmp(ts->name, "_EndCount")) {
-        endCountType = ts->type;
+    // Look for the type var `_remoteEndCountType` in ChapelBase.
+    forv_Vec(VarSymbol, var, gVarSymbols) {
+      const char* searchStr = "_remoteEndCountType";
+      if (strcmp(var->cname, searchStr) == 0) {
+        endCountType = var->type;
```

Comment:
As in a PrimitiveType that we create within the compiler? Possibly, especially now that we want different behavior for cases the compiler understands well...

I think it will be best to go with the current structure for this release. I will add some comments suggesting that this may be worth looking into.